### PR TITLE
Add optional layout property to suppress loading of Bootstrap 3 grid

### PR DESCRIPTION
**Pull request by uhafner**
Jenkins bundles the grid of Bootstrap 3 (that has been patched to use 24 columns). It is not compatible with new JS frameworks (see https://issues.jenkins-ci.org/browse/JENKINS-61326) that are based on BS4 and a 12 column grid (see https://github.com/jenkinsci/bootstrap4-api-plugin).

This pull request provides a workaround by adding a new optional property to the `layout.jelly` file until a fix for https://issues.jenkins-ci.org/browse/JENKINS-61326 is available. Using this toggle a plugin can suppress loading of the BS3 grid when using the layout tag. 

### Proposed changelog entries

Developer: Add `nogrid` option to `layout.jelly` tag to allow suppressing the bootstrap 3 grid. See [bootstrap4-api-plugin](https://github.com/jenkinsci/bootstrap4-api-plugin) for details.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [X] (If applicable) Jira issue is well described
- [X] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@fqueiruga 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by fqueiruga — CHANGES_REQUESTED** [review]
Hi Ulli, there's some CSS code that needs to be extracted off the `responsive-grid.less`.

- All rules from the beginning until to https://github.com/jenkinsci/jenkins/blob/ab5616712ab8d79bcabc9a97e10a0edb2f5696c4/war/src/main/webapp/css/responsive-grid.css#L28
- All rules from this one till the end https://github.com/jenkinsci/jenkins/blob/ab5616712ab8d79bcabc9a97e10a0edb2f5696c4/war/src/main/webapp/css/responsive-grid.css#L1714

I also think the `hidden-*` classes need to be backported, as they are used on the header & footer. Otherwise, we can just manually change the `layout.jelly` and `pageHeader.jelly` to remove them, but that may be a bit more work.

**Review by timja — APPROVED** [review]

**Comment by fqueiruga** [comment]
@uhafner I have created a PR against your branch that migrates essential CSS off the `responsive-grid.css` file: https://github.com/uhafner/jenkins/pull/1.

I have only done a smoke test on the `hidden-*` ones and they work, but it be good to have more confirmation. I haven't found a single use of the `visible-*` classes across the whole ecosystem, but I think it's best to keep them for now.

Please let us know once there's an artifact of the warnings-ng plugin with these changes, I'll make it a priority to test it.

Also, can you create an issue to this one? I think it's a long shot that this is backported but it would be really nice.

**Comment by uhafner** [comment]
I can split the grid css into the suggested parts, but shouldn't this be in a separate PR?

**Comment by fqueiruga** [comment]
I can create the PR with the changes if you prefer. The other PR should be merged before this one then.

**Comment by uhafner** [comment]
For me it makes no difference. If it is ok from a core perspective then I can merge it here. 

BTW: I created an additional issue: https://issues.jenkins-ci.org/browse/JENKINS-62094?filter=-2

**Comment by fqueiruga** [comment]
Maybe it's indeed best to keep both changes separated. I'll keep the other PR Today.

**Review by fqueiruga — APPROVED** [review]
Approving it but I think it should not be merged before https://github.com/jenkinsci/jenkins/pull/4697

**Comment by timja** [comment]
I think this might be worth a developer changelog entry?

**Comment by uhafner** [comment]
Here in core as well? (I will definitely describe the usage in the bootstrap API plugin).

**Comment by timja** [comment]
Just a minor one liner I think yes

**Comment by timja** [comment]
Something like:
> Developer: Add `nogrid` option to layout.jelly to allow suppressing the bootstrap 3 grid

**Comment by uhafner** [comment]
Ok, done.

**Comment by timja** [comment]
Let's merge this after ~24 hours, (it should be merged together with https://github.com/jenkinsci/jenkins/pull/4697)

**Review by fcojfernandez — APPROVED** [review]

**Review by varyvol — APPROVED** [review]

**Comment by daniel-beck** [comment]
And once we no longer do this anyway, this would become a no-op? Or what is the long-term plan here? Is it even supported to `nogrid=false`?

**Comment by uhafner** [comment]
The long term plan is to remove the flag (or make it non functional) when we replace the Bootstrap 3 grid. Once this PR has been merged I can set this flag to `true` in the jelly tag of the bootstrap plugin. So actually no other plugin should care about this flag at all.

**Pull request merged**
